adders/prettier/config/adder.ts

```diff
@@ -0,0 +1,128 @@
+import { dedent, defineAdderConfig } from "@svelte-add/core";
+import { options } from "./options.js";
+
+export const adder = defineAdderConfig({
+    metadata: {
+        id: "prettier",
+        name: "Prettier",
+        description: "An opinionated code formatter",
+        environments: { svelte: true, kit: true },
+        website: {
+            logo: "./prettier.svg",
+            keywords: ["prettier", "code", "formatter", "formatting"],
+            documentation: "https://prettier.io",
+        },
+    },
+    options,
+    integrationType: "inline",
+    packages: [
+        { name: "prettier", version: "^3.3.2", dev: true },
+        { name: "prettier-plugin-svelte", version: "^3.2.5", dev: true },
+        {
+            name: "eslint-config-prettier",
+            version: "^9.1.0",
+            dev: true,
+            condition: ({ dependencies }) => hasEslint(dependencies),
+        },
+    ],
+    files: [
+        {
+            name: () => `.prettierignore`,
+            contentType: "text",
+            content: ({ content }) => {
+                if (content) return content;
+                return dedent`
+                    # Package Managers
+                    package-lock.json
+                    pnpm-lock.yaml
+                    yarn.lock
+                `;
+            },
+        },
+        {
+            name: () => ".prettierrc",
+            contentType: "json",
+            content: ({ data }) => {
+                data.useTabs ??= true;
```

Comment:
I'm not sure we should set these properties if the user already has a `.prettierrc`. It essentially means they've setup prettier already and we're modifying their config. I think we could add `prettier-plugin-svelte` in that case, but I don't think I'd change any formatting options

Maybe this section should be more like:
```
if (!Object.keys(data).length) {
  data.useTabs = true;
  data.singleQuote = true;
  data.trailingComma = 'none';
  data.printWidth = 100;
}
```

Reply:
Isn't that code exactly doing this? At least from my c# backgound i would assume that this only assigns useTabs if useTabs is null / undefined

Reply:
> I'm not sure we should set these properties if the user already has a `.prettierrc`. It essentially means they've setup prettier already and we're modifying their config. I think we could add `prettier-plugin-svelte` in that case, but I don't think I'd change any formatting options

Ah this is an excellent point, nice catch!

> Isn't that code exactly doing this? At least from my c# backgound i would assume that this only assigns useTabs if useTabs is null / undefined

yes and no. yes that it only assigned `useTabs` if it's `undefined`, but if they already have a prettier config configured, we don't want to add new rules to it (with the exception of adding the `prettier-plugin-svelte`)

Reply:
> this only assigns useTabs if useTabs is null / undefined

that's correct

> Isn't that code exactly doing this?

not quite. E.g. if the user has setup a `.prettierrc` which only specifies `singleQuote = true` and they've left `useTabs` empty, then we would come along and set `useTabs = true` and I'm not sure that's what we want